deltachat-ios/Chat/ChatViewController.swift

```diff
@@ -529,12 +529,22 @@ class ChatViewController: UITableViewController {
         let id = messageIds[indexPath.row]
         if id == DC_MSG_ID_DAYMARKER {
             let cell = tableView.dequeueReusableCell(withIdentifier: "info", for: indexPath) as? InfoMessageCell ?? InfoMessageCell()
-            if messageIds.count > indexPath.row + 1 {
-                let nextMessageId = messageIds[indexPath.row + 1]
-                let nextMessage = dcContext.getMessage(id: nextMessageId)
-                cell.update(text: DateUtils.getDateString(date: nextMessage.sentDate))
-            } else {
-                cell.update(text: "ErrDaymarker")
+            var stop = false
+            var n = 1
+            // pick up the date of the next received, sent or info message
+            while !stop {
+                if messageIds.count > indexPath.row + n {
+                    let nextMessageId = messageIds[indexPath.row + n]
+                    if nextMessageId != DC_MSG_ID_MARKER1 {
+                        let nextMessage = dcContext.getMessage(id: nextMessageId)
+                        cell.update(text: DateUtils.getDateString(date: nextMessage.sentDate))
+                        stop = true
+                    }
+                    n += 1
+                } else {
+                    cell.update(text: "ErrDaymarker")
+                    stop = true
+                }
```

Comment:
i am a bit sceptical using a loop here, that may easily go wrong and we would have a deadlock (off by one? end conditon never set? things may change in the future and result in errors?) 

as we do accept everything but DC_MSG_ID_MARKER1 for  the date, and there is only one DC_MSG_ID_MARKER1 by definition, however, a loop seems not to be needed.

i would suggest to sth. as https://github.com/deltachat/deltachat-ios/pull/1280 therefore.

Reply:
true, I was not sure about info messages in first place, but I left them out in the end